### PR TITLE
Hide p2p debug output, and use logger to log

### LINKED ISSUE #20: Hide debug output from p2p sub-package
https://github.com/eoscanada/eos-go/blob/f320fb22d1a5c19ad599d7ee82d7ef6ba31372c1/p2p/client.go#L153

https://github.com/eoscanada/eos-go/blob/f320fb22d1a5c19ad599d7ee82d7ef6ba31372c1/p2p/client.go#L155

https://github.com/eoscanada/eos-go/blob/f320fb22d1a5c19ad599d7ee82d7ef6ba31372c1/p2p/client.go#L161

https://github.com/eoscanada/eos-go/blob/f320fb22d1a5c19ad599d7ee82d7ef6ba31372c1/p2p/client.go#L212-L214

etc

It's making usage of package frustrating

### PR DESCRIPTION
**Pull request by fanyang1988**
fix #20 Hide debug output from p2p sub-package
delete all print in p2p package
add a logger interface base on zap.SugaredLogger to p2p,
then use logger to log, this can allow developer use some other logger in project base on eos-go/p2p

### CODE REVIEW

File: p2p/logger.go [hunk 1]
```diff
@@ -0,0 +1,72 @@
+package p2p
+
+import "go.uber.org/zap"
+
+// SugaredLoggerInterface a interface for zap sugared logger,
+// use this can allow some project based on eos-go use it's different logger imp
+type SugaredLoggerInterface interface {
+
```
Comment (abourget) [inline p2p/logger.go]: Why import the interface if we're already dependent on zap ?

**Comment by abourget** [comment]
We already wired in zap in the "eos-go" package. Perhaps we can reuse the same patterns.

Also, if possible, let's not use the Sugar interface and use the base structured logging. Systems using the p2p might want to ensure maximal performance.

What do you think?

File: p2p/logger.go (hunk 1 above)
Comment (fanyang1988) [reply p2p/logger.go]: at first i wannot to change the log format in p2p package, just hide, so a interface can allow someone to use diff logger, like this : [https://github.com/uber-go/zap/blob/master/FAQ.md#why-arent-logger-and-sugaredlogger-interfaces](url), but if we use structured logging by zap, i think it is no need to add a interface.

**Comment by fanyang1988** [comment]
the new commit use structured log by zap, and the p2pLog in p2p package is just imp by the same way to eos-go package, i think the next step is to make a external package to process loggers, and allow developer can config zap logger, and it will be a independent PR

File: p2p/logger.go (hunk 1 above)
Comment (abourget) [reply p2p/logger.go]: It's such a huge Interface that I think it's not that useful in the end. I'd prefer having only an io.Writer as what is passed in by users of this lib. The zap.Logger can then use that as its output.

**Comment by abourget** [comment]
Ok it's a lot better now.

Thanks for the contribution!

**Pull request merged**